decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -12,30 +23,161 @@ class GlobalHandler(Handler):
     # Dict translating endianness between the binaryninja enum and pythons literals
     Endian = {Endianness.LittleEndian: "little", Endianness.BigEndian: "big"}
 
+    def __init__(self, lifter):
+        super().__init__(lifter)
+        self._datavariable_types = {
+            CharType: self._lift_basic_type,
+            IntegerType: self._lift_basic_type,
+            FloatType: self._lift_basic_type,
+            BoolType: self._lift_basic_type,
+            VoidType: self._lift_void_type,
+            ArrayType: self._lift_constant_type,
+            PointerType: self._lift_pointer_type,
+        }
+
     def register(self):
         """Register the handler at its parent lifter."""
         self._lifter.HANDLERS.update({DataVariable: self.lift_global_variable})
 
     def lift_global_variable(self, variable: DataVariable, view: BinaryView, 
-        parent: Optional[MediumLevelILInstruction] = None, **kwargs
-    ) -> UnaryOperation:
-        """Lift global variables with basic types (pointer are possible)"""
+        parent: Optional[MediumLevelILInstruction] = None, caller_addr: int = 0, **kwargs
+    ) -> Union[Symbol, UnaryOperation, GlobalVariable, StringSymbol]:
+        """Lift global variables via datavariable type. Check bninja error case + recursive datavariable first"""
+        if not self._addr_in_section(view, variable.address):
+            return Constant(variable.address, vartype=Integer(view.address_size*8, False))
+
+        if caller_addr == variable.address: 
+            return self._lifter.lift(variable.symbol) if variable.symbol else \
+            Symbol("data_" + f"{variable.address:x}", variable.address, vartype=Integer.uint32_t())
+
+        return self._datavariable_types[type(variable.type)](variable, view, parent)
+
+
+    def _lift_constant_type(self, variable: DataVariable, view: BinaryView, parent: Optional[MediumLevelILInstruction] = None) -> StringSymbol:
+        """Lift a constant datavariable directly into code (bninja should only yield strings)"""
+        string = str(variable.value)[2:-1].rstrip('\\x00')
```

Comment:
variable.value is a `bytes` therefore it has the `b"..."` between the real value. 
The constant values simply purge them. 
The strip on the right side is to remove NULL chars from the string. 

Therefore if we have a string like: "Hello World\nHi\x00" we convert it to "Hello World\nHi". 
We said one time that we want to keep control chars like newline/tab etc, but purge the Null byte at the end. 

(Sometimes bninja actually gives more then 1 Nullbyte at the end, therefore we strip every null byte at the right side.

Reply:
Ok, but I would add it in a short from into comment. Otherwise if Binja at some point changes this behavior, it would take quite a while to trace it to the right place.

Reply:
Done